File: Python/18-github-copilot-app/sprint_planner.py
"""Sprint-planning workstream for the Copilot desktop app demo."""

from work_items import SprintPlan, WorkItem


class SprintPlanner:
    """Builds a priority-ordered plan within a sprint capacity."""

    def __init__(self, items: list[WorkItem]) -> None:
        if items is None:
            raise TypeError("items cannot be None")
        self._items = tuple(items)

    def build_plan(self, capacity: int) -> SprintPlan:
        """Return ready work selected within the available capacity."""
        # TODO (Copilot app, workstream A): implement the behavior in TASK.md.
        raise NotImplementedError

File: Javascript/18-github-copilot-app/sprintPlanner.js
'use strict';

class SprintPlanner {
  /**
   * @param {{ id: string, title: string, priority: string, status: string, estimate: number, createdAt: Date, dueAt: Date | null }[]} items
   */
  constructor(items) {
    if (!Array.isArray(items)) {
      throw new TypeError('items must be an array');
    }

    this.items = [...items];
  }

  /**
   * Select ready work within the available capacity.
   * @param {number} capacity
   * @returns {{ items: object[], totalEstimate: number }}
   */
  buildPlan(capacity) {
    // TODO (Copilot app, workstream A): implement the behavior in TASK.md.
    throw new Error('Not implemented');
  }
}

module.exports = { SprintPlanner };